components/script/dom/range.rs

```diff
@@ -1189,9 +1189,8 @@ impl WeakRangeVec {
                 let move_start = node_is_start && range.start_offset() == offset;
                 let move_end = node_is_end && range.end_offset() == offset;
 
-                let remove_from_node = move_start && move_end ||
-                    move_start && !node_is_end ||
-                    move_end && !node_is_start;
+                let remove_from_node =
+                    !(!move_start || !move_end && node_is_end) || move_end && !node_is_start;
```

Comment:
Hrm...I think this new formulation is more difficult to understand. In this case, I would add an exception for the clippy warning or rewrite the logic to be more understandable.

Reply:
I absolutely agree with you @mrobinson ...I picked this straight from the clippy suggestion and felt it was more complicated but thought it was the standard. I checked that it evaluated to the same as the before and applied it.
Let me find another way for this sir.
Thank you.